lib/s3_helper.rb

```diff
@@ -23,10 +23,11 @@ def initialize(s3_bucket_name, tombstone_prefix = '', options = {})
   end
 
   def upload(file, path, options = {})
+    options[:body] = file
     path = get_path_for_s3_upload(path)
     obj = s3_bucket.object(path)
-    obj.upload_file(file, options)
-    path
+    output = obj.put(options)
```

Comment:
This makes me very uncomfortable. We are losing the multipart upload for files > 15MB by using `put` instead of `upload_file` and this restricts the maximum upload size to 5GB. That might be too low for backups.

I understand why you are doing it. It's very unfortunate that `upload_file` does return a boolean instead of the server's response. Still, I think the better option is to use `FileUploader` even though it's marked as "private api". Hopefully our tests will fail when the API changes.

https://github.com/aws/aws-sdk-ruby/blob/f465e9a7500659b74fce7d1cc4f17e6616aa0a4c/gems/aws-sdk-s3/lib/aws-sdk-s3/customizations/object.rb#L305-L308

Reply:
yes I already added the workaround https://github.com/discourse/discourse/commit/1e547c0ebc46a11115bf873e41c9af797d3ac881

Reply:
What are the trade-offs using `obj.load` here?

Reply:
Multipart upload won't return etag (just void). So we using `.load` method to do http `head` request. It will update the object data except the body (file content).

Reply:
Ahh icic. So we’ll need to do an extra request for large uploads. I guess for multipart uploads that isn’t really a huge deal since we’re already doing multiple requests.